wrapanapi/systems/ec2.py

```diff
@@ -1187,3 +1246,45 @@ def get_registry_data(self):
                     'registry': first_registry['proxyEndpoint']}
         except (IndexError, KeyError):
             raise NotFoundError("couldn't get registry details. please check environment setup")
+
+    def create_network(self, cidr_block='10.0.0.0/16'):
+        try:
+            response = self.ec2_connection.create_vpc(CidrBlock=cidr_block)
+            network_id = response.get('Vpc').get('VpcId')
+            return EC2Vpc(system=self, uuid=network_id, raw=self.ec2_resource.Vpc(network_id))
+        except Exception:
+            return False
+
+    def get_network(self, name):
+        networks = self.find_networks(name)
+        if not networks:
+            raise NotFoundError("Network with name {} not found".format(name))
+        elif len(networks) > 1:
+            raise MultipleItemsError("Multiple networks with name {} found".format(name))
+        return networks[0]
+
+    def list_networks(self):
+        """
+        Returns a list of Network objects
+        """
+        network_list = [
+            EC2Vpc(system=self, uuid=vpc['VpcId'], raw=self.ec2_resource.Vpc(vpc['VpcId']))
+            for vpc in self.ec2_connection.describe_vpcs().get('Vpcs')
+        ]
+        return network_list
+
+    def find_networks(self, name_or_id):
+        """
+        Return list of all networks with given name or id
+        Args:
+            name_or_id: name or id to search for if starts with vpc- then it's evaluated as id
+        Returns:
+            List of EC2Vpc objects
+        """
+        if name_or_id.startswith('vpc-'):
```

Comment:
I'm actually thinking it would be better to split this into two kwargs, although its not necessarily as clean.

I don't think we can assume that the caller passing `'vpc-.*'` isn't searching for a network with a tagged name that starts with 'vpc-'.

Reply:
Again, I used it from find_templates method where is it the same: https://github.com/ManageIQ/wrapanapi/blob/master/wrapanapi/systems/ec2.py#L740
I can change it of course.

Reply:
I don't see how find_templates is the same? Its taking two separate kwargs, one specifically for `id` and one specifically for `name`. Its not making an assumption of name or id based on the value of the param.

Reply:
Reworked. It now requires either name or id.